Documentation/ApiOverview/Database/RestrictionBuilder/Index.rst

```diff
@@ -199,32 +199,110 @@ QueryRestrictionContainer
+
     This container is always added if not told otherwise.
 
 :php:`\TYPO3\CMS\Core\Database\Query\Restriction\FrontendRestrictionContainer`
-    Adds :php:`\TYPO3\CMS\Core\Database\Query\Restriction\DeletedRestriction`,
-    :php:`\TYPO3\CMS\Core\Database\Query\Restriction\HiddenRestriction`,
-    :php:`\TYPO3\CMS\Core\Database\Query\Restriction\StartTimeRestriction`,
-    :php:`\TYPO3\CMS\Core\Database\Query\Restriction\EndTimeRestriction`,
-    :php:`\TYPO3\CMS\Core\Database\Query\Restriction\WorkspaceRestriction`
-    and :php:`\TYPO3\CMS\Core\Database\Query\Restriction\FrontendGroupRestriction`.
-    This container should be be added by a developer to a query when creating
+    Adds
+
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\DeletedRestriction`
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\HiddenRestriction`
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\StartTimeRestriction`
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\EndTimeRestriction`
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\WorkspaceRestriction`
+    -   :php:`\TYPO3\CMS\Core\Database\Query\Restriction\FrontendGroupRestriction`
+
+    This container should be added by a developer to a query when creating
     query statements in frontend context or when handling frontend stuff from
-    within CLI calls.
+    within :ref:`CLI <symfony-console-commands>` calls.
 
-limitRestrictionsToTables()
-===========================
+:php:`\TYPO3\CMS\Core\Database\Query\Restriction\LimitToTablesRestrictionContainer`
+    This restriction container applies added restrictions only to the given
+    table aliases. See :ref:`database-limit-restrictions-to-tables` for more
+    information. Enforced restrictions are treated equally to all other
+    restrictions.
 
-With the :php:`\TYPO3\CMS\Core\Database\Query\Restriction\LimitToTablesRestrictionContainer`
+..  _database-limit-restrictions-to-tables:
+
+Limit restrictions to tables
+============================
+
+With :php:`\TYPO3\CMS\Core\Database\Query\Restriction\LimitToTablesRestrictionContainer`
 it is possible to apply restrictions to a query only for a given set of tables,
 or - to be precise - table aliases. Since it is a restriction container, it can
 be added to the restrictions of the query builder and can hold restrictions
 itself.
 
+Examples
+--------
+
+If you want to apply one or more restrictions to only one table, that is
+possible as follows. Let's say you have content in the :sql:`tt_content` table
+with a relation to categories. Now you want to get all records with their
+categories except those that are hidden. In this case, the hidden restriction
+should apply only to the :sql:`tt_content` table, not to the :sql:`sys_category`
+or :sql:`sys_category_*_mm` table.
+
+
+..  code-block:: php
+    :caption: EXT:some_extension/Classes/Domain/Repository/ContentRepository.php
+
+    $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
```

Comment:
Can we move this example into a file and maybe inject the connection pool?

Reply:
Yup, thought also about it. But then this should be done with all examples. So in a separate PR?

Reply:
I would convert code snippets whenever you touch them as you also have to apply linting and code style then to pass the tests

Reply:
Yup, in the next part. Will provide a follow-up PR (with all examples) to not clutter this one.

Reply:
ok

Reply:
ok